File: OpenCV4 基本プログラミング/OpenCV4 基本プログラミング/src/UMat.cpp
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

void my_gray_scale(UMat src) {
    UMat dst;
    cvtColor(src, dst, COLOR_RGB2GRAY);
    
    imshow("src", src);
    imshow("dst", dst);
    
    waitKey(0);
    return;
}

File: OpenCV4-BasicProgramming/src/Chapter3/UMat.py
import cv2


def main(arg: str):
    src: cv2.UMat
    dst: cv2.UMat
    src = cv2.imread(arg)
    dst = cv2.cvtColor(src, cv2.COLOR_RGB2GRAY)
    cv2.imshow("src", src)
    cv2.imshow("dst", dst)
    cv2.waitKey(0)
    return


if __name__ == '__main__':
    main()
